staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1319,6 +1323,17 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				return []watch.Event{{Type: watch.Added, Object: createdPodsAfterWatch[0]}}
 			},
 		},
+		{
+			name:                               "sendInitialEvents=false, RV=useCurrentRV",
```

Comment:
Can you additionally add a similar case (useCurrentRV) for 
a) sendInitialEvents=true 
b) sendInitialEvents=true & allwWatchBookmarks=true

[the one you added looks good already]

Reply:
Actually I started wondering if we need the case I added. 
How does it differ from 
 - `sendInitialEvents=false, RV=1` 
- or the one you proposed `b` how does it differ from `sendInitialEvents=true & allowWatchBookmarks=true, RV=1`.

In both cases `RV=1` and `RV=useCurrentRV` are behind "the initial events". 

I think the `useCurrentRV` could be read after adding the initial events and that would give us a new flavour. WDYT?

I have prepared the following matrix (I will add it to the source code) for better visualising the current state and potential gaps in testing.

```
// RunWatchSemantics test the following cases:
//
// +-----------------+---------------------+-------------------+
// | ResourceVersion | AllowWatchBookmarks | SendInitialEvents |
// +-----------------+---------------------+-------------------+
// | Unset           | -                   | -/-               |
// | Unset           | false               | true/false        |
// | 0               | true                | true/-            |
// | 0               | false               | true/false        |
// | 1               | true                | true/-            |
// | 1               | false               | true/-            |
// +-----------------+---------------------+-------------------+
//
// where:
// - false indicates the value of the param was set to "false" by a test case
// - true  indicates the value of the param was set to "true" by a test case
```


The first case is interesting and I'm surprised we have missed it. But then I realised that we have `RunWatchSemanticInitialEventsExtended` which actually covers `allowWatchBookmarks=true, sendInitialEvents=true, RV=unset`. We moved it to a separate test because it ensure the global rv will be delivered as part of the initial bookmarks. Which is fine but I think we could add a standard case here as well.

Reply:
> I think the useCurrentRV could be read after adding the initial events and that would give us a new flavour.

Agree.

Reply:
Are you also OK with adding more cases to fill the matrix ?

Reply:
pushed, PTAL.